src/connection_string.ts

```diff
@@ -631,7 +634,28 @@ export const OPTIONS = {
     target: 'credentials',
     transform({ options, values: [value] }): MongoCredentials {
       if (typeof value === 'string') {
-        value = toRecord(value);
+        const validKeys = [
+          'SERVICE_NAME',
+          'SERVICE_REALM',
+          'CANONICALIZE_HOST_NAME',
+          'AWS_SESSION_TOKEN'
+        ];
+
+        const mechanismProperties = Object.create(null);
+
+        for (const [key, _value] of entriesFromString(value)) {
+          if (validKeys.includes(key)) {
+            if (key === 'CANONICALIZE_HOST_NAME') {
+              mechanismProperties[key] = getBoolean(key, _value);
```

Comment:
Just a note that this is probably going to conflict with Durran's PR to expand the CANONICALIZE_HOST_NAME options